subsys/bluetooth/host/gatt.c

```diff
@@ -180,13 +183,125 @@ BT_GATT_SERVICE_DEFINE(_2_gap_svc,
 #endif
 );
 
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
-static void sc_ccc_cfg_changed(const struct bt_gatt_attr *attr,
-			       u16_t value)
+struct sc_data {
+	u16_t start;
+	u16_t end;
+} __packed;
+
+struct gatt_sc_cfg {
+	u8_t		id;
```

Comment:
id and peer addr is duplicated in gatt_sc_cfg and sc_ccc.cfg. Could we maybe just have an array of `struct sc_data`, and use the same index to save some context data?

Reply:
If you mean we should add a "struct sc_data" to bt_gatt_ccc_cfg, the issue is that we add (4 * BT_GATT_CCC_MAX) bytes in every defined CCC attribute and not only the one used for SC, which can add up very quickly. The drawback of course is that it requires additional lookups when we need to do something with SC data.

Maybe in the future we could use the connection index and bonded device index (in keys array) instead of  having to do lookups using id and address.

Reply:
No, that's not what i meant. I meant to remove those two parameters, and use the same index into both gatt_scf_cfg and sc_ccc.cfg to link the information. Just a thought.

Reply:
Ah ok, I see what you meant! It's a nice idea, but if we later have CCC loaded on demand (and I would like to work on that soon), we will only be left with CCC data for connected peers and we will then miss indexes for bonded devices, which we will most likely still need for SC data.
I'll see if something like that can be done once I get to it, thanks for the suggestion!